Source/NexusForever.WorldServer/Game/Social/GlobalChatManager.cs

```diff
@@ -340,40 +343,45 @@ ChatResult GetResult()
         public void HandleWhisperChat(WorldSession session, ClientChatWhisper whisper)
         {
             Player target = CharacterManager.Instance.GetPlayer(whisper.PlayerName);
-            if (target == null)
-            {
-                SendMessage(session, $"Player \"{whisper.PlayerName}\" not found.");
-                return;
-            }
 
-            if (session.Player.Name == target.Name)
+            bool CanWhisper()
             {
-                SendMessage(session, "You cannot send a message to yourself.");
-                return;
+                if (target == null)
+                    return false;
+
+                if (session.Player.Name == target.Name)
+                    return false;
```

Comment:
I lack of knowledge C# but I think this is simpler
```suggestion
                if (target == null || session.Player.Name == target.Name)
                    return false;
```